R/missingDataImputation.R

```diff
@@ -254,6 +275,50 @@ MissingDataImputation <- function(jaspResults, dataset, options) {
 
 ###------------------------------------------------------------------------------------------------------------------###
 
+.loggedEventsToTable <- function(jaspResults, options) {
+  miceMids <- jaspResults[["MiceMids"]]
+  miceOut <- miceMids$object
+  events <- miceOut$loggedEvents
+
+  if (is.null(jaspResults[["LoggedEventsTable"]])) {
+    table <- createJaspTable("Logged events")
+    table$dependOn(c("method", "nImps", "nIter", "seed", "printAllLoggedEvents", "maxLoggedEvents", "passive", "passiveImputation", "quickpred", "imputationVariables"))
+    table$addColumnInfo(name = "Iteration", title = "Iteration", type = "integer")
+    table$addColumnInfo(name = "Imputation", title = "Imputation", type = "integer")
+    table$addColumnInfo(name = "Variable", title = "Variable", type = "string")
+    table$addColumnInfo(name = "Message", title = "Message", type = "string")
+    table$addColumnInfo(name = "Predictor", title = "Predictor", type = "string")
+    jaspResults[["LoggedEventsTable"]] <- table
+  } else {
+    table <- jaspResults[["LoggedEventsTable"]]
+  }
+
+  if (is.null(events) || nrow(events) == 0) {
+    table$addFootnote(paste0("No events were logged."))
+    return()
+  }
+
+  maxToShow <- if (isTRUE(options$printAllLoggedEvents)) nrow(events) else options$maxLoggedEvents
+  nShown <- min(nrow(events), maxToShow)
+  for (i in seq_len(nShown)) {
+    table$addRows(list(
+      Iteration = events[i, "it"],
+      Imputation = events[i, "im"],
+      Variable = events[i, "dep"],
+      Message = events[i, "meth"],
+      Predictor = events[i, "out"]
+    ))
+  }
+
+  if (nShown < nrow(events)) {
+    table$addFootnote(paste0("Showing ", nShown, " of a total of ", nrow(events), " events"))
```

Comment:
Consider aligning and simplifying these statements. E.g.,
"Showing XX of YY events."
"Showing all events."